lib/fizzy/stack.hpp

```diff
@@ -4,39 +4,85 @@
 
 #pragma once
 
+#include <cassert>
 #include <cstdint>
+#include <memory>
 #include <vector>
 
 namespace fizzy
 {
-template <typename T>
-class Stack : public std::vector<T>
```

Comment:
Can we keep `Stack`, because it is now used in #312?
Or what can be the other way forward?

Reply:
I will drop the commit removing it, but I also changed the interface. Is it fine, or you would like to keep the original one?

Reply:
I think it's fine, I'll rebase on new interface.

Reply:
Done.